optimum/habana/transformers/generation/utils.py

```diff
@@ -471,7 +471,7 @@ def _prepare_generated_length(
         return generation_config
 
     def _prepare_generation_config(
-        self, generation_config: GaudiGenerationConfig, lazy_mode: bool, **kwargs: Dict
+        self, generation_config: GaudiGenerationConfig, **kwargs: Dict
```

Comment:
The lazy_mode variable is not propogating downstream after this function call.
Replacing the function signature as in the original Transformer (https://github.com/huggingface/transformers/blob/4fdf58afb72b0754da30037fc800b6044e7d9c99/src/transformers/generation/utils.py#L1231) fixes this problem.

Reply:
@MohitIntel  can you check how it was working on main branch(transformer 4.38) without any issue?

Reply:
This function is originally called in https://github.com/huggingface/transformers/blob/main/src/transformers/models/whisper/generation_whisper.py#L484 with only two parameters : generation_config and **kwargs.
The lazy_mode variable is part of the kwargs dict in this call.
But, in the transformer_4.40 branch, this lazy_mode variable is getting **unpacked as a separate variable and gets popped from the kwargs dict.** 
Thus, it does not propagate downstream as part of kwargs in https://github.com/huggingface/transformers/blob/60bb571e993b7d73257fb64044726b569fef9403/src/transformers/models/whisper/generation_whisper.py#L591 as seen here from the debug logs:

transformers_4.40 **kwargs:
```python
(Pdb) {'labels': tensor([[50276, 50359, 50363,  ...,  -100,  -100,  -100],
        [50276, 50359, 50363,  ...,  -100,  -100,  -100],
        [50276, 50359, 50363,  ...,  -100,  -100,  -100],
        ...,
        [50276, 50359, 50363,  ...,  -100,  -100,  -100],
        [50276, 50359, 50363,  ...,  -100,  -100,  -100],
        [50276, 50359, 50363,  ...,  -100,  -100,  -100]], device='hpu:0'), 'hpu_graphs': True}
``` 
 v/s Fixed version **kwargs which contains the **'lazy_mode': True key-value pair** :
```python
 (Pdb) {'labels': tensor([[50276, 50359, 50363,  ...,  -100,  -100,  -100],
        [50276, 50359, 50363,  ...,  -100,  -100,  -100],
        [50276, 50359, 50363,  ...,  -100,  -100,  -100],
        ...,
        [50276, 50359, 50363,  ...,  -100,  -100,  -100],
        [50276, 50359, 50363,  ...,  -100,  -100,  -100],
        [50276, 50359, 50363,  ...,  -100,  -100,  -100]], device='hpu:0'), 'hpu_graphs': True, 'lazy_mode': True}
```